components/script/dom/gpuadapter.rs

```diff
@@ -60,4 +71,47 @@ impl GPUAdapterMethods for GPUAdapter {
     fn Extensions(&self, _cx: SafeJSContext) -> NonNull<JSObject> {
         NonNull::new(self.extensions.get()).unwrap()
     }
+
+    /// https://gpuweb.github.io/gpuweb/#dom-gpuadapter-requestdevice
+    fn RequestDevice(&self, descriptor: &GPUDeviceDescriptor, comp: InCompartment) -> Rc<Promise> {
+        let promise = Promise::new_in_current_compartment(&self.global(), comp);
+        let sender = response_async(&promise, self);
+        let id = self.global().as_window().Navigator().create_device_id();
```

Comment:
I just noticed that `GPUAdapter` is possibly exposed to `DedicatedWorker`.

Maybe we should call these window methods inside a `if let` instead 👀?

Ex.

```rust
if let Some(window) = self.global().downcast::<Window>() {
    let id = window.Navigator().create_device_id();
    ...
}
```

Reply:
Thank you! I will updated the pull request with this change.